src/main/java/eu/darkbot/impl/managers/AttackApiImpl.java

```diff
@@ -0,0 +1,116 @@
+package eu.darkbot.impl.managers;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.objects.facades.SettingsProxy;
+import eu.darkbot.api.entities.other.Ammo;
+import eu.darkbot.api.entities.utils.Attackable;
+import eu.darkbot.api.managers.AttackAPI;
+import eu.darkbot.api.managers.HeroAPI;
+import eu.darkbot.api.managers.HeroItemsAPI;
+import eu.darkbot.api.objects.Item;
+import org.jetbrains.annotations.Nullable;
+
+public class AttackApiImpl implements AttackAPI {
+
+    private final SettingsProxy settingsProxy;
+    private final HeroItemsAPI heroItems;
+    private final HeroAPI hero;
+
+    private Attackable target;
+    private long lockTryTime, attackTryTime;
+
+    public AttackApiImpl(SettingsProxy settingsProxy,
+                         HeroItemsAPI heroItems,
+                         HeroAPI hero) {
+        this.settingsProxy = settingsProxy;
+        this.heroItems = heroItems;
+        this.hero = hero;
+    }
+
+    @Override
+    public @Nullable Attackable getTarget() {
+        return target;
+    }
+
+    @Override
+    public @Nullable Attackable setTarget(@Nullable Attackable attackable) {
```

Comment:
not too sure about what the utility of previous target really is. If one place wants to use it, it can use getTarget before calling setTarget, i wouldn't return it every time and have to do the extra variable here